### ISSUE TITLE
eaf-open-terminal immediately terminates

### ISSUE THREAD

**Issue by cmower** [manateelazycat/emacs-application-framework#458]
**Describe the bug**
When I call `eaf-open-terminal` the process fails and when I use a minimal `init.el` nothing works.

Using my normal initialization (I use spacemacs), running `M-x eaf-open-terminal` results in the following in the `*Messages*` buffer:
```
[EAF] Process starting...
[EAF/terminal] Opening ~/
[EAF] *eaf* killed
[EAF] Process terminated.
```

The output in the `*eaf*` buffer is
```
127.0.0.1 - - [30/Nov/2020 20:47:40] "GET /index.html HTTP/1.1" 200 -
Session is not restored, as no data present in session file.
127.0.0.1 - - [30/Nov/2020 20:47:40] "GET /node_modules/xterm/lib/xterm.js HTTP/1.1" 200 -
127.0.0.1 - - [30/Nov/2020 20:47:40] "GET /node_modules/xterm-addon-attach/lib/xterm-addon-attach.js HTTP/1.1" 200 -
127.0.0.1 - - [30/Nov/2020 20:47:40] "GET /node_modules/xterm-addon-fit/lib/xterm-addon-fit.js HTTP/1.1" 200 -
127.0.0.1 - - [30/Nov/2020 20:47:40] "GET /node_modules/xterm-addon-search/lib/xterm-addon-search.js HTTP/1.1" 200 -
127.0.0.1 - - [30/Nov/2020 20:47:40] "GET /node_modules/xterm-addon-web-links/lib/xterm-addon-web-links.js HTTP/1.1" 200 -
127.0.0.1 - - [30/Nov/2020 20:47:40] "GET /dark_theme.js HTTP/1.1" 200 -
127.0.0.1 - - [30/Nov/2020 20:47:40] "GET /node_modules/xterm/css/xterm.css HTTP/1.1" 200 -
js: WebSocket connection to 'ws://127.0.0.1:55485/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
```

I tried a minimal initialization, installed ```use-package``` manually from MELPA and set 
```elisp
(use-package eaf
  :load-path "/home/chris/Software/emacs-application-framework"
  :custom
  (eaf-find-alternate-file-in-dired t)
  :config
  (eaf-bind-key scroll_up "C-n" eaf-pdf-viewer-keybinding)
  (eaf-bind-key scroll_down "C-p" eaf-pdf-viewer-keybinding)
  (eaf-bind-key take_photo "p" eaf-camera-keybinding))
```
as `init.el`. In this case, `eaf.el` fails during initialization. The output from the `*Backtrace*` is
```
Debugger entered--Lisp error: (file-missing "Cannot open load file" "No such file or directory" "s")
  require(s)
  eval-buffer(#<buffer  *load*-8994> nil "/home/chris/Software/emacs-application-framework/eaf.el" nil t)  ; Reading at buffer position 2240
  load-with-code-conversion("/home/chris/Software/emacs-application-framework/eaf.el" "/home/chris/Software/emacs-application-framework/eaf.el" t t)
  require(eaf nil t)
  (not (require 'eaf nil t))
  (if (not (require 'eaf nil t)) (display-warning 'use-package (format "Cannot load %s" 'eaf) :error) (condition-case err (progn (eaf-bind-key scroll_up "C-n" eaf-pdf-viewer-keybinding) (eaf-bind-key scroll_down "C-p" eaf-pdf-viewer-keybinding) (eaf-bind-key take_photo "p" eaf-camera-keybinding) t) ((debug error) (funcall use-package--warning0 :config err))))
  (progn (funcall (or (get 'eaf-find-alternate-file-in-dired 'custom-set) #'set-default) 'eaf-find-alternate-file-in-dired t) (if (not (require 'eaf nil t)) (display-warning 'use-package (format "Cannot load %s" 'eaf) :error) (condition-case err (progn (eaf-bind-key scroll_up "C-n" eaf-pdf-viewer-keybinding) (eaf-bind-key scroll_down "C-p" eaf-pdf-viewer-keybinding) (eaf-bind-key take_photo "p" eaf-camera-keybinding) t) ((debug error) (funcall use-package--warning0 :config err)))))
  (condition-case err (progn (funcall (or (get 'eaf-find-alternate-file-in-dired 'custom-set) #'set-default) 'eaf-find-alternate-file-in-dired t) (if (not (require 'eaf nil t)) (display-warning 'use-package (format "Cannot load %s" 'eaf) :error) (condition-case err (progn (eaf-bind-key scroll_up "C-n" eaf-pdf-viewer-keybinding) (eaf-bind-key scroll_down "C-p" eaf-pdf-viewer-keybinding) (eaf-bind-key take_photo "p" eaf-camera-keybinding) t) ((debug error) (funcall use-package--warning0 :config err))))) ((debug error) (funcall use-package--warning0 :catch err)))
  eval-buffer(#<buffer  *load*> nil "/home/chris/.emacs.d/init.el" nil t)  ; Reading at buffer position 703
  load-with-code-conversion("/home/chris/.emacs.d/init.el" "/home/chris/.emacs.d/init.el" t t)
  load("/home/chris/.emacs.d/init" noerror nomessage)
  startup--load-user-init-file(#f(compiled-function () #<bytecode 0x15705a240b95>) #f(compiled-function () #<bytecode 0x15705a240ba9>) t)
  command-line()
  normal-top-level()
```

My clone of `emacs-application-framework` is on the `master` branch and up-to-date. 

**Expected behavior**
A terminal to be opened when calling `eaf-open-terminal`. 

Apologies if I've missed something basic in the setup/configuration. As far as is listed in `README.md` I have installed all required dependancies. 

**Versions (please complete the following info):**
 - Distro and DE/WM: Ubuntu 20.04, GNOME
 - Versions of Dependencies:
    * PyQt5: 5.15.2
    * PyQt5-sip: 12.8.1
    * dbus-python: 1.2.16
    * PyQtWebEngine: 5.15.2
    * wmctrl: 1.07
    * NodeJS: v10.19.0
 - M-x emacs-version: GNU Emacs 27.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.20)

**Comment by manateelazycat** [comment]
look https://github.com/manateelazycat/emacs-application-framework/issues/452

**Comment by cmower** [comment]
Thanks for your suggestion, it has changed the behaviour but unfortunately this didn't solve my issue. 

As suggested in #452, I tried running `node server.js` in `app/terminal/` first and saw similar output
```
innerError { Error: Cannot find module '../build/Debug/pty.node'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty/lib/unixTerminal.js:30:15)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3) code: 'MODULE_NOT_FOUND' }
/home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty/lib/unixTerminal.js:35
        throw outerError;
        ^

Error: The module '/home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty/build/Release/pty.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 88. This version of Node.js requires
NODE_MODULE_VERSION 64. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:807:18)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty/lib/unixTerminal.js:26:11)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
```

Then I ran `npm rebuild`, not sure if it's relevant but the output from this is as follows
```

> node-pty@0.9.0 install /home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty
> node scripts/install.js

make: Entering directory '/home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty/build'
  CXX(target) Release/obj.target/pty/src/unix/pty.o
In file included from ../src/unix/pty.cc:20:
../../nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../../nan/nan.h:2298:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
 2298 |     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
      |                                                              ^
../src/unix/pty.cc: In function ‘void pty_after_waitpid(uv_async_t*)’:
../src/unix/pty.cc:486:43: warning: ‘void* memset(void*, int, size_t)’ writing to an object of type ‘class Nan::Persistent<v8::Function>’ with no trivial copy-assignment [-Wclass-memaccess]
  486 |   memset(&baton->cb, -1, sizeof(baton->cb));
      |                                           ^
In file included from ../../nan/nan.h:403,
                 from ../src/unix/pty.cc:20:
../../nan/nan_persistent_12_inl.h:12:40: note: ‘class Nan::Persistent<v8::Function>’ declared here
   12 | template<typename T, typename M> class Persistent :
      |                                        ^~~~~~~~~~
In file included from ../../nan/nan.h:54,
                 from ../src/unix/pty.cc:20:
../src/unix/pty.cc: At global scope:
/usr/include/nodejs/src/node.h:573:43: warning: cast between incompatible function types from ‘void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’ {aka ‘void (*)(v8::Local<v8::Object>)’} to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
  573 |       (node::addon_register_func) (regfunc),                          \
      |                                           ^
/usr/include/nodejs/src/node.h:607:3: note: in expansion of macro ‘NODE_MODULE_X’
  607 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
      |   ^~~~~~~~~~~~~
../src/unix/pty.cc:708:1: note: in expansion of macro ‘NODE_MODULE’
  708 | NODE_MODULE(pty, init)
      | ^~~~~~~~~~~
In file included from /usr/include/nodejs/src/node.h:63,
                 from ../../nan/nan.h:54,
                 from ../src/unix/pty.cc:20:
/usr/include/nodejs/deps/v8/include/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]’:
/usr/include/nodejs/src/node_object_wrap.h:84:78:   required from here
/usr/include/nodejs/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<node::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
 9502 |                reinterpret_cast<Callback>(callback), type);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/nodejs/deps/v8/include/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]’:
../../nan/nan_object_wrap.h:65:61:   required from here
/usr/include/nodejs/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
  SOLINK_MODULE(target) Release/obj.target/pty.node
  COPY Release/pty.node
make: Leaving directory '/home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty/build'

> node-pty@0.9.0 postinstall /home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty
> node scripts/post-install.js

node-pty@0.9.0 /home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty
nan@2.14.0 /home/chris/Software/emacs-application-framework/app/terminal/node_modules/nan
ws@7.2.3 /home/chris/Software/emacs-application-framework/app/terminal/node_modules/ws
xterm@4.4.0 /home/chris/Software/emacs-application-framework/app/terminal/node_modules/xterm
xterm-addon-attach@0.5.0 /home/chris/Software/emacs-application-framework/app/terminal/node_modules/xterm-addon-attach
xterm-addon-fit@0.3.0 /home/chris/Software/emacs-application-framework/app/terminal/node_modules/xterm-addon-fit
xterm-addon-search@0.5.0 /home/chris/Software/emacs-application-framework/app/terminal/node_modules/xterm-addon-search
xterm-addon-web-links@0.2.1 /home/chris/Software/emacs-application-framework/app/terminal/node_modules/xterm-addon-web-links
```

Then I tried running `eaf-open-terminal` in emacs. It seems the terminal opens but I can't interact with it, there is just a blank screen, see below.

![Screenshot from 2020-12-03 11-30-10](https://user-images.githubusercontent.com/8171456/101013556-67393d80-355c-11eb-8715-0f5e875bba5c.png)

I ran `node server.js` in `app/terminal/` again and got the following output
```
/home/chris/Software/emacs-application-framework/app/terminal/server.js:14
var commands = process.argv.slice(4)[0].split(" ")
                                       ^

TypeError: Cannot read property 'split' of undefined
    at Object.<anonymous> (/home/chris/Software/emacs-application-framework/app/terminal/server.js:14:40)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
```

**Comment by manateelazycat** [comment]
Because you need pass more argument to server.js, you can't call "node server.js"

**Comment by manateelazycat** [comment]
Can you try "node install node-pty" or "node rebuild node-pty", error message is clear, node-pty version need upgrade.

**Comment by cmower** [comment]
> Because you need pass more argument to server.js, you can't call "node server.js"

Apologies, I am not familiar with nodejs/javascript/etc at all. 

> Can you try "node install node-pty" or "node rebuild node-pty", error message is clear, node-pty version need upgrade.

When in `/app/terminal`:

```
$ node install node-pty
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module '/home/chris/Software/emacs-application-framework/app/terminal/install'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)

$ node rebuild node-pty
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module '/home/chris/Software/emacs-application-framework/app/terminal/rebuild'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)

```

**Comment by manateelazycat** [comment]
sorry，try 

npm install node-pty

**Comment by cmower** [comment]
> sorry，try
> 
> npm install node-pty

No problem, output in terminal (while in `/app/terminal`) is
```
$ npm install node-pty
npm WARN checkPermissions Missing write access to /home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty/node_modules/nan
npm WARN eaf-terminal@1.0.0 No description
npm WARN eaf-terminal@1.0.0 No repository field.

npm ERR! code ENOENT
npm ERR! syscall access
npm ERR! path /home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty/node_modules/nan
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, access '/home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty/node_modules/nan'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/chris/.npm/_logs/2020-12-03T16_34_55_299Z-debug.log

$ npm rebuild node-pty

> node-pty@0.9.0 install /home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty
> node scripts/install.js

make: Entering directory '/home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty/build'
  CXX(target) Release/obj.target/pty/src/unix/pty.o
In file included from ../src/unix/pty.cc:20:
../../nan/nan.h: In function ‘void Nan::AsyncQueueWorker(Nan::AsyncWorker*)’:
../../nan/nan.h:2298:62: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
 2298 |     , reinterpret_cast<uv_after_work_cb>(AsyncExecuteComplete)
      |                                                              ^
../src/unix/pty.cc: In function ‘void pty_after_waitpid(uv_async_t*)’:
../src/unix/pty.cc:486:43: warning: ‘void* memset(void*, int, size_t)’ writing to an object of type ‘class Nan::Persistent<v8::Function>’ with no trivial copy-assignment [-Wclass-memaccess]
  486 |   memset(&baton->cb, -1, sizeof(baton->cb));
      |                                           ^
In file included from ../../nan/nan.h:403,
                 from ../src/unix/pty.cc:20:
../../nan/nan_persistent_12_inl.h:12:40: note: ‘class Nan::Persistent<v8::Function>’ declared here
   12 | template<typename T, typename M> class Persistent :
      |                                        ^~~~~~~~~~
In file included from ../../nan/nan.h:54,
                 from ../src/unix/pty.cc:20:
../src/unix/pty.cc: At global scope:
/usr/include/nodejs/src/node.h:573:43: warning: cast between incompatible function types from ‘void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’ {aka ‘void (*)(v8::Local<v8::Object>)’} to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
  573 |       (node::addon_register_func) (regfunc),                          \
      |                                           ^
/usr/include/nodejs/src/node.h:607:3: note: in expansion of macro ‘NODE_MODULE_X’
  607 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
      |   ^~~~~~~~~~~~~
../src/unix/pty.cc:708:1: note: in expansion of macro ‘NODE_MODULE’
  708 | NODE_MODULE(pty, init)
      | ^~~~~~~~~~~
In file included from /usr/include/nodejs/src/node.h:63,
                 from ../../nan/nan.h:54,
                 from ../src/unix/pty.cc:20:
/usr/include/nodejs/deps/v8/include/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = node::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)]’:
/usr/include/nodejs/src/node_object_wrap.h:84:78:   required from here
/usr/include/nodejs/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<node::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<node::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
 9502 |                reinterpret_cast<Callback>(callback), type);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/nodejs/deps/v8/include/v8.h: In instantiation of ‘void v8::PersistentBase<T>::SetWeak(P*, typename v8::WeakCallbackInfo<P>::Callback, v8::WeakCallbackType) [with P = Nan::ObjectWrap; T = v8::Object; typename v8::WeakCallbackInfo<P>::Callback = void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)]’:
../../nan/nan_object_wrap.h:65:61:   required from here
/usr/include/nodejs/deps/v8/include/v8.h:9502:16: warning: cast between incompatible function types from ‘v8::WeakCallbackInfo<Nan::ObjectWrap>::Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<Nan::ObjectWrap>&)’} to ‘Callback’ {aka ‘void (*)(const v8::WeakCallbackInfo<void>&)’} [-Wcast-function-type]
  SOLINK_MODULE(target) Release/obj.target/pty.node
  COPY Release/pty.node
make: Leaving directory '/home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty/build'

> node-pty@0.9.0 postinstall /home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty
> node scripts/post-install.js

node-pty@0.9.0 /home/chris/Software/emacs-application-framework/app/terminal/node_modules/node-pty

```

Restarted emacs, and tried `eaf-open-terminal`, same behaviour as before (terminal opens but I cannot interact with anything, it is just a blank screen), output from `*eaf*` is 
```
EAF process starting...
1607013350.2279122 75 PyQt5.QtCore.QRect(0, 0, 640, 480)
1607013350.2293065 217 PyQt5.QtCore.QRect(0, 0, 640, 480)
1607013350.2294025 203 PyQt5.QtCore.QRect(0, 0, 640, 480)
1607013350.2294312 34 PyQt5.QtCore.QRect(0, 0, 640, 480)
1607013350.2294729 13 PyQt5.QtCore.QRect()
1607013350.2295015 14 PyQt5.QtCore.QRect()
1607013350.2295215 17 PyQt5.QtCore.QRect()
1607013350.22956 26 PyQt5.QtCore.QRect()
1607013350.2507024 68 PyQt5.QtCore.QRect()
127.0.0.1 - - [03/Dec/2020 16:35:50] "GET /index.html HTTP/1.1" 200 -
1607013350.2788877 68 PyQt5.QtCore.QRect()
1607013350.2791853 21 PyQt5.QtCore.QRect()
1607013350.2792597 15 PyQt5.QtCore.QRect()
1607013350.2792933 109 PyQt5.QtCore.QRect()
1607013350.2794044 152 PyQt5.QtCore.QRect()
1607013350.2804024 75 PyQt5.QtCore.QRect()
1607013350.2804604 69 PyQt5.QtCore.QRect()
1607013350.2805183 13 PyQt5.QtCore.QRect()
1607013350.2805617 14 PyQt5.QtCore.QRect()
1607013350.2806087 17 PyQt5.QtCore.QRect()
1607013350.2833576 26 PyQt5.QtCore.QRect()
1607013350.2851467 13 PyQt5.QtCore.QRect(0, 0, 2560, 1379)
1607013350.2852232 14 PyQt5.QtCore.QRect(0, 0, 2560, 1379)
1607013350.2936656 14 PyQt5.QtCore.QRect(0, 0, 2560, 1379)
Session is not restored, as no data present in session file.
1607013350.2946005 74 PyQt5.QtCore.QRect(0, 0, 2560, 1379)
1607013350.294717 74 PyQt5.QtCore.QRect(0, 0, 2560, 1379)
1607013350.294747 43 PyQt5.QtCore.QRect(0, 0, 2560, 1379)
1607013350.2947595 76 PyQt5.QtCore.QRect(0, 0, 2560, 1379)
1607013350.294774 78 PyQt5.QtCore.QRect(0, 0, 2560, 1379)
1607013350.2947876 77 PyQt5.QtCore.QRect(0, 0, 2560, 1379)
127.0.0.1 - - [03/Dec/2020 16:35:50] "GET /node_modules/xterm/lib/xterm.js HTTP/1.1" 200 -
1607013350.3631434 13 PyQt5.QtCore.QRect(0, 0, 2560, 1396)
1607013350.363261 14 PyQt5.QtCore.QRect(0, 0, 2560, 1396)
1607013350.369536 14 PyQt5.QtCore.QRect(0, 0, 2560, 1396)
127.0.0.1 - - [03/Dec/2020 16:35:50] "GET /node_modules/xterm-addon-attach/lib/xterm-addon-attach.js HTTP/1.1" 200 -
127.0.0.1 - - [03/Dec/2020 16:35:50] "GET /node_modules/xterm/css/xterm.css HTTP/1.1" 200 -
127.0.0.1 - - [03/Dec/2020 16:35:50] "GET /node_modules/xterm-addon-fit/lib/xterm-addon-fit.js HTTP/1.1" 200 -
127.0.0.1 - - [03/Dec/2020 16:35:50] "GET /node_modules/xterm-addon-search/lib/xterm-addon-search.js HTTP/1.1" 200 -
127.0.0.1 - - [03/Dec/2020 16:35:50] "GET /node_modules/xterm-addon-web-links/lib/xterm-addon-web-links.js HTTP/1.1" 200 -
1607013350.3732936 77 PyQt5.QtCore.QRect(0, 0, 2560, 1396)
127.0.0.1 - - [03/Dec/2020 16:35:50] "GET /dark_theme.js HTTP/1.1" 200 -
js: WebSocket connection to 'ws://127.0.0.1:57113/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
1607013350.534495 2 PyQt5.QtCore.QRect(0, 0, 2560, 1396)
1607013354.127275 13 PyQt5.QtCore.QRect(0, 0, 1280, 1404)
1607013354.1273913 14 PyQt5.QtCore.QRect(0, 0, 1280, 1404)
1607013354.1368341 14 PyQt5.QtCore.QRect(0, 0, 1280, 1404)
1607013354.137509 77 PyQt5.QtCore.QRect(0, 0, 1280, 1404)
1607013354.2578464 13 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013354.2625535 14 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013354.2721303 14 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013354.2824316 77 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013354.3137968 183 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013354.3139043 183 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013355.6880703 13 PyQt5.QtCore.QRect(0, 0, 1280, 818)
1607013355.6881948 14 PyQt5.QtCore.QRect(0, 0, 1280, 818)
1607013355.693459 14 PyQt5.QtCore.QRect(0, 0, 1280, 818)
1607013355.694205 77 PyQt5.QtCore.QRect(0, 0, 1280, 818)
1607013357.1893797 13 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013357.189441 14 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013357.1901827 14 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013357.1904042 77 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013388.677948 13 PyQt5.QtCore.QRect(0, 0, 1280, 1243)
1607013388.6780198 14 PyQt5.QtCore.QRect(0, 0, 1280, 1243)
1607013388.6802862 14 PyQt5.QtCore.QRect(0, 0, 1280, 1243)
1607013388.6805356 77 PyQt5.QtCore.QRect(0, 0, 1280, 1243)
1607013388.7410185 13 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013388.7410855 14 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013388.7415276 14 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
1607013388.7417562 77 PyQt5.QtCore.QRect(0, 0, 1280, 1396)
```

Am I running this in the correct directory? I'm not sure if this matters for nodejs...

**Comment by manateelazycat** [comment]
Hmm, looks is permission problem.

Can you try "sudo npm install node-pty" and "sudo npm rebuild node-pty" ?

**Comment by manateelazycat** [comment]
Just close issue, please re-open issue if have any further clue.

**Comment by cmower** [comment]
I was able to fix the issue by upgrading NodeJS from `v10.19.0` to `v14.15.1`, then rebuilding as in #452.
